### PR TITLE
Initial work on CPU specific registers

### PR DESCRIPTION
**Pull request by XVilka**

### CODE REVIEW

**Comment by Maijin** [comment]
Doesn't build yet, r_db is deprecated afaik thoughhttps://travis-ci.org/radare/radare2/jobs/274016628#L728

**Comment by XVilka** [comment]
I pushed it as is, will update asap.

**Comment by MaskRay** [comment]
Why another register type? 

off-topic, i think functions in r_reg.h should be reduced

**Comment by Maijin** [comment]
@MaskRay https://github.com/radare/radare2/issues/8467

File: binr/rasm2/rasm2.c [hunk 1]
```diff
@@ -602,6 +602,8 @@ int main (int argc, char *argv[]) {
 	r_asm_set_cpu (a, cpu);
 	r_asm_set_bits (a, (env_bits && *env_bits)? atoi (env_bits): bits);
 	r_anal_set_bits (anal, (env_bits && *env_bits)? atoi (env_bits): bits);
+	a->sysregs = r_sysregs_new ();
+	r_sysregs_setup(a->sysregs, arch, cpu);
```
Comment (radare) [inline binr/rasm2/rasm2.c]: missing space

File: libr/core/cconfig.c [hunk 2]
```diff
@@ -463,6 +464,12 @@ static int cb_asmarch(void *user, void *data) {
 			r_config_set_i (core->config, "asm.pcalign", 0);
 		}
 	}
+	if (cpu = r_config_get (core->config, "anal.cpu")) {
+		if (!r_sysregs_setup(core->anal->sysregs, asmos, cpu)) {
```
Comment (radare) [inline libr/core/cconfig.c]: missing space

File: libr/syscall/sysregs.c [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	file = sdb_fmt (0, "%s/%s-%s-%d.sdb",
+		SYSREGSPATH, arch, cpu);
+	if (!r_file_exists (file)) {
+		eprintf ("r_sysregs_setup: Cannot find '%s'\n", file);
+	}
+
+	sdb_close (s->db);
+	sdb_free (s->db);
+	s->db = sdb_new (0, file, 0);
+	if (s->fd)
```
Comment (radare) [inline libr/syscall/sysregs.c]: missing braces

**Comment by radare** [comment]
pls fix those tiny changes and ill merge. toomanyopenprs

**Comment by radare** [comment]
ping

**Comment by XVilka** [comment]
Will finish it this weekend, totally forgot about this one.

**Comment by radare** [comment]
Cool. Yeah its been standing here for so much time. I will try to close mines too

**Comment by Maijin** [comment]
Would be nice to have an example of sdb

**Comment by Maijin** [comment]
ping

**Comment by Maijin** [comment]
ping2 ?

**Comment by radare** [comment]
Ping3

**Comment by Maijin** [comment]
https://github.com/fluxchief/binaryninja_avr/tree/master/chips

**Pull request closed without merging**